src/Quantum/Quantum.jl

```diff
@@ -9,19 +9,17 @@ Tensor Network [`Ansatz`](@ref) that has a notion of sites and directionality (i
 """
 abstract type Quantum <: Ansatz end
 
-function checkmeta(::Type{Quantum}, tn::TensorNetwork)
-    # meta exists
-    haskey(tn.metadata, :plug) || return false
+metadata(::Type{Quantum}) = NamedTuple{(:plug,),Tuple{Dict{Tuple{Int,Symbol},Symbol}}}
 
+function checkmeta(::Type{Quantum}, tn::TensorNetwork)
     # meta has correct type
-    tn[:plug] isa AbstractDict{Tuple{Int,Symbol},Symbol} || return false
-    all(∈(:in, :out) ∘ last, keys(tn[:plug])) || return false
+    all(∈((:in, :out)) ∘ last, keys(tn.plug)) || return false
```

Comment:
From the way that we defined `metadata`, I think that we need to access to `plug` via `tn.metadata.plug`
```suggestion
    all(∈((:in, :out)) ∘ last, keys(tn.metadata.plug)) || return false
```

Reply:
1. I already fixed it in a commit that I haven't uploaded yet. I will upload it in a second.
2. Nope! I've implemented `getproperty!` and `propertynames!` for `TensorNetwork` so `tn.plug` is valid. Nice eh?

Reply:
I will take a look at the code after your commits.